src/App.tsx

```diff
@@ -15,47 +16,127 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = '',
+  Alphabetically = 'alphabetically',
+  ByLength = 'length',
+}
+
+enum SortOrder {
+  Default = 0,
+  Ascending = 1,
+  Descending = -1,
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortType: SortType,
+  isReversed: SortOrder,
+): string[] {
+  const preparedGoods: string[] = [...goods];
+
+  if (sortType !== SortType.Default) {
+    preparedGoods.sort((item1, item2) => {
+      if (sortType === SortType.Alphabetically) {
+        return item1.localeCompare(item2);
+      }
+
+      if (sortType === SortType.ByLength) {
+        return item1.length - item2.length;
+      }
+
+      return 0;
+    });
+  }
+
+  if (isReversed === SortOrder.Ascending) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+function changeAscDesc(value: SortOrder): SortOrder {
+  let typeOfOrder = value;
+
+  if (typeOfOrder === SortOrder.Ascending) {
+    typeOfOrder = SortOrder.Descending;
+  } else {
+    typeOfOrder = SortOrder.Ascending;
+  }
+
+  return typeOfOrder;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.Default);
+  const [isReversed, setIsReversed] = useState(SortOrder.Default);
+  const isResetButtonVisible = sortType || isReversed === SortOrder.Ascending;
+
+  const sortedGoods: string[]
+    = getPreparedGoods(goodsFromServer, sortType, isReversed);
+
+  function handleButtonReverse(): void {
+    setSortType(SortType.Default);
+    setIsReversed(SortOrder.Default);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button is-info',
+            { 'is-light': sortType !== SortType.Alphabetically },
+          )}
```

Comment:
```suggestion
          className={classNames(
            'button',
            'is-info',
            { 'is-light': sortType !== SortType.Alphabetically },
          )}
```

Same in other places